### ISSUE TITLE
update_node module ability to update polling method to SNMPv3

### ISSUE THREAD

**Issue by Andyjb8** [jeisenbath/ansible-collection-solarwinds-orion#14]
Can the  solarwinds.orion.orion_update_node module update a node that is currently using polling method of ICMP to SNMPv3? 

These fields didn't seem to be valid under properties

```
- name: Update node to SNMP polling
  solarwinds.orion.orion_update_node:
    hostname: "{{ solarwinds_server }}"
    username: "{{ solarwinds_user }}"
    password: "{{ solarwinds_pass }}"
    name: "{{ node_name }}"
    properties:
      polling_method: SNMP
      snmp_version: 3
      snmpv3_auth_key: "{{ snmpv3_auth_pass}}" 
      snmpv3_auth_method: SHA1
      snmpv3_priv_key: "{{ snmpv3_priv_pass}}"
      snmpv3_priv_method: AES128
      snmpv3_username:  "{{ snmpv3_user}}"
  delegate_to: localhost
```

**Comment by jeisenbath** [comment]
The examples in the module should reflect it, but the "properties" you pass are the fields in the Orion database. One of the examples listed is for updating to snmpv2, I haven't worked out what fields you'd need to update one to snmp v3

**Comment by jeisenbath** [comment]
Just following up on this, were you ever able to work out the properties or is this something that still needs to be worked out? I do believe its a good example to add to documentation.

**Comment by Andyjb8** [comment]
I just open [PR 28](https://github.com/jeisenbath/ansible-collection-solarwinds-orion/pull/28)

I realized I didn't test to see if it impacted updating other proprieties besides SMMPv3. 

Also, if node is already running SNMPv3 I think we would want it to update the SNMPv3 credentials, as that could be used if you wanted to change them. However I was getting a error as I stated in PR.

**Comment by jeisenbath** [comment]
So this is turning out a little tricky actually, I'll see if I can explain it well.

The function you introduced in the PR, properties_need_update(), was actually only checking against the custom orion.get_node() which only returns a specific set of properties. To correct that, I added logic to query for the params passed.

The issue then is that the SNMPv3 properties for auth and private key aren't in Orion.Nodes, so I attempted to query them from Orion.SNMPv3Credentials table by adding a table parameter to the module...except the table property names in that table don't match the properties you need to update. On top of that even taking out the call to the properties_need_update() function to allow it to update, my updated node is still failing to poll. 

I don't know, ultimately, I think this is going to be not possible, and the best way to update a node to SNMPv3 would be to remove it and add it back. I'm sure its possible to make the right API calls to update a node to SNMPv3, but this wouldn't be the correct module for it. I do like the function you put in, that will improve the module's idempotence.

**Comment by jeisenbath** [comment]
Ok, I actually did get it working (it was missing two properties...and I also was testing on an unconfigured device oops), and I committed the example to the devel 2.1.x branch. I had to take out the function properties_need_update() to get this to work, since those properties are in orion.node and I couldn't think of a generic way to make that work.

I noted in the example, but updating from ICMP to SNMP will of course also require tasks to add the correct SNMP pollers. I stand by that in the case of updating from ICMP to SNMP, it is likely best to remove and readd the node, however this should work fine when going from SNMPv2 to SNMPv3 (and I may use this myself now!)

**Comment by Andyjb8** [comment]
I have another use case to change/update SNMPv3 credentials. Can this module also update the SNMPv3 credentials of a node in Solarwinds? I tested it and it didn't work. Is it possible that this module could support that capability as well?

**Comment by jeisenbath** [comment]
Released the 2.1.0 version #41 which had the example that should show how to do this. For your question on updating the snmpv3 credentials, no this module can't do that. I have compiled a 'wish list' that I'm putting into a discussion for the next release though, so I'll put that use case into that list.
